beacon-chain/types/active_state.go

```diff
@@ -228,8 +251,33 @@ func (a *ActiveState) CalculateNewActiveState(
 		}
 	}
 
+	_, proposerIndex, err := casper.ProposerShardAndIndex(
+		cState.ShardAndCommitteesForSlots(),
+		cState.LastStateRecalculationSlot(),
+		parentSlot)
+	if err != nil {
+		return nil, fmt.Errorf("Can not get proposer index %v", err)
+	}
+
+	a.setRandaoMix(block.RandaoReveal())
+
+	specialRecordData := make([][]byte, 2)
+	for i := range specialRecordData {
+		specialRecordData[i] = make([]byte, 32)
+	}
+	blockRandao := block.RandaoReveal()
+	binary.PutUvarint(specialRecordData[0], proposerIndex)
+	specialRecordData[1] = blockRandao[:]
```

Comment:
Yeah. I have to initialize the nested slice otherwise I get `panic: runtime error: index out of range` error. Let me know if there's a better way of doing it.
```
nested := make([][]byte, 2)
for i := range nested {
		nested[i] = make([]byte, 32)
	}
nested[0] = "bleh bleh"
nested[1] = "lalala"
```

Reply:
What I think you're describing is legal: https://play.golang.org/p/M2kxdaJmbTu

are you referring to the line above, `binary.PutUvarint(specialRecordData[0], proposerIndex)`? 

if so, you should be able to initialize just the first index.